subprojects/core/src/main/java/org/gradle/api/internal/plugins/DefaultObjectConfigurationAction.java

```diff
@@ -97,9 +102,14 @@ private void applyScript(Object script) {
         ScriptSource scriptSource = new UriScriptSource("script", scriptUri);
         ClassLoaderScope classLoaderScopeChild = classLoaderScope.createChild("script-" + scriptUri.toString());
         ScriptHandler scriptHandler = scriptHandlerFactory.create(scriptSource, classLoaderScopeChild);
-        ScriptPlugin configurer = configurerFactory.create(scriptSource, scriptHandler, classLoaderScopeChild, classLoaderScope, false);
-        for (Object target : targets) {
-            configurer.apply(target);
+        final ScriptPlugin configurer = configurerFactory.create(scriptSource, scriptHandler, classLoaderScopeChild, classLoaderScope, false);
+        for (final Object target : targets) {
+            buildOperationExecutor.run("Apply " + scriptSource.getDisplayName() + " to " + target, new Action<BuildOperationContext>() {
```

Comment:
this should be inside the `apply()` method, as this should just be a fundamental part of applying a script plugin, regardless of where and to what we're applying the script.